hazelcast/src/main/java/com/hazelcast/spi/impl/merge/AbstractPartitionedContainerCollector.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.merge;
+
+import com.hazelcast.spi.NodeEngine;
+
+import java.util.Collection;
+
+/**
+ * Base implementation of {@link AbstractContainerCollector} for partitioned data structures.
+ *
+ * @param <C> container of the data structure
+ */
+public abstract class AbstractPartitionedContainerCollector<C> extends AbstractContainerCollector<C> {
```

Comment:
Yeah, the naming of the collectors is misleading. We should change this. Maybe something like `AbstractPartitionContainerCollector` and `AbstractNamedContainerCollector` fits better. And internally use `hasIterator()` instead of `isPartitioned()` to distinct (which also gives a better hint which methods are used).

I'll also have a look if we can use iterators for the named containers.

Reply:
I've implemented the iterator for the `AbstractNamedContainerCollector` (needed a little tweak, I think you version never made progress again since `advance()` was just called once). So the `AbstractPartitionContainerCollector` could be completely removed (together with a bunch of methods).